platforms/core-configuration/model-core/src/main/java/org/gradle/api/internal/provider/CollectionPropertyExtensions.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2025 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.internal.provider;
+
+import org.gradle.api.provider.HasMultipleValues;
+import org.gradle.api.provider.Provider;
+
+@SuppressWarnings("unused") // registered as Groovy extension in ExtensionModule
+public class CollectionPropertyExtensions {
+
+    /**
+     * Adds an element to the property value
+     *
+     * <p>Extension method to support the left shift operator in Groovy.</p>
+     *
+     * @param self the {@link org.gradle.api.provider.ListProperty} or the {@link org.gradle.api.provider.SetProperty}
+     * @param element the element
+     */
+    public static <T> void leftShift(HasMultipleValues<T> self, T element) {
+        self.add(element);
+    }
+
+    /**
+     * Adds a provider of the element to the property value
+     *
+     * <p>Extension method to support the left shift operator in Groovy.</p>
+     *
+     * @param self the {@link org.gradle.api.provider.ListProperty} or the {@link org.gradle.api.provider.SetProperty}
+     * @param provider the provider of the element
+     */
+    public static <T> void leftShift(HasMultipleValues<T> self, Provider<? extends T> provider) {
+        self.add(provider);
+    }
+
+    /**
+     * Adds elements to the property value
+     *
+     * <p>Extension method to support the left shift operator in Groovy.</p>
+     *
+     * @param self the {@link org.gradle.api.provider.ListProperty} or the {@link org.gradle.api.provider.SetProperty}
+     * @param elements the elements
+     */
+    public static <T> void leftShift(HasMultipleValues<T> self, Iterable<? extends T> elements) {
```

Comment:
❓ In Groovy you cannot use `<<` to append multiple elements. I'm not sure if we should support this for List/Set properties.

Reply:
Why we test it tho?
https://github.com/gradle/gradle/blob/cd2331a1a3668117106748402100e1e812f174e5/platforms/core-configuration/model-core/src/integTest/groovy/org/gradle/api/provider/GroovyPropertyAssignmentIntegrationTest.groovy#L188

Reply:
>Why we test it tho?

To define the behavior and make sure it fails in the expected manner. There are plenty of things we don't plan to support there.